module/VuFind/src/VuFind/Search/SearchOrigin/AbstractSearchOrigin.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Search Origin Object
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  TODO?
+ * @author   Robby ROUDON <roudonro@msu.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Search\SearchOrigin;
+
+/**
+ * Abstract object for any search origin
+ *
+ * @category VuFind
+ * @package  TODO?
```

Comment:
It probably makes sense to use Search here. Most of the VuFind\Search namespace already uses the Search package, and I don't think there's a strong reason to break out SearchOrigin separately.